crates/ra_syntax/src/ast/expr_extensions.rs

```diff
@@ -305,44 +305,34 @@ impl ast::Literal {
             .unwrap()
     }
 
-    pub fn kind(&self) -> LiteralKind {
-        match self.token().kind() {
-            INT_NUMBER => {
-                let int_suffix_list = [
-                    "isize", "i128", "i64", "i32", "i16", "i8", "usize", "u128", "u64", "u32",
-                    "u16", "u8",
-                ];
+    fn find_suffix(text: &str, possible_suffixes: &[&str]) -> Option<SmolStr> {
+        possible_suffixes
+            .iter()
+            .find(|&suffix| text.ends_with(suffix))
+            .map(|&suffix| SmolStr::new(suffix))
+    }
 
-                // The lexer treats e.g. `1f64` as an integer literal. See
-                // https://github.com/rust-analyzer/rust-analyzer/issues/1592
-                // and the comments on the linked PR.
-                let float_suffix_list = ["f32", "f64"];
+    pub fn kind(&self) -> LiteralKind {
+        const INT_SUFFIXES: [&'static str; 12] = [
```

Comment:
FYI: There is a caveat.
If the language introduces new suffixes or user-defined suffixes (who knows)
You should ensure that suffixes in these arrays do not include each other or replace .ends_with()
test with something more elaborated (or just order the suffixes properly to avoid a bug)